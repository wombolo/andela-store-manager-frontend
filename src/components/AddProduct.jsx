import React, { Component } from 'react';
import {addNewProduct} from '../actions/productActions';
import PropTypes from "prop-types";
import {connect} from "react-redux";

export class AddProduct extends Component {
  state={
    image_file:'',
    title:'',
    price:'',
    quantity:'',
    description:'',
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.addNewProduct(this.state);
  };

  handleChange = (event) => {
    this.setState({[event.target.id]: event.target.value});
  };

  render() {
    return (
      <div>
        <h1>Add New Product</h1>

        <div className="row clearfix products-grid">
          <div className="col-10 m-0-auto m-t-5">
            <h3>Single Product Page</h3>
            <form onSubmit={this.handleSubmit}>

              <div className="row">

                <div className="col-3">
                  <figure className="image-holder">
                    <img alt={'image'} src="../assets/images/products/product-1.png" />
                    <input type={'file'} id={'image_file'}
                           onChange={ (e) => this.handleChange({
                             target:{
                               id:'image_file',
                               value: e.target.files[0],
                             }
                           })
                           }/>
                  </figure>
                </div>


                <div className="col-7">
                  <table border="0" width="100%" className="product-single-table table-striped">
                    <tbody>
                    <tr>
                      <td><label htmlFor="title">Name</label></td>
                      <td><input id="title" type="text" required={true} onChange={this.handleChange}/></td>
                    </tr>
                    <tr>
                      <td><label htmlFor="description">Description</label></td>
                      <td><textarea rows={15} id="description" required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label htmlFor="price">Price</label></td>
                      <td><input id="price" type="number" step={'.01'} required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label htmlFor="quantity">Quantity Available</label></td>
                      <td><input id="quantity" type={'number'} required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label>Product Category</label></td>
                      <td><select>
                        <option>Choose One</option>
                        <option>Fiction/ Sci-Fi</option>
                        <option>Comedy</option>
                        <option>Romance</option>
                        <option>Adventure</option>
                        <option>Thriller</option>
                        <option>Drama</option>
                      </select></td>
                    </tr>

                    <tr>
                      <td>&nbsp;</td>
                      <td><button type="submit" className="btn btn-primary">Save</button>
                        <button type="reset" className="btn btn-danger">Clear</button></td>
                    </tr>
                    </tbody>

                  </table>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    )
  }
}

AddProduct.propTypes = {
  addNewProduct: PropTypes.func.isRequired
};

export default connect(null, {
  addNewProduct,
})(AddProduct);
