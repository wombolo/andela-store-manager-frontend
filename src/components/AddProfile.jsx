import React, { Component } from 'react';
import {addNewProfile} from '../actions/profileActions';
import PropTypes from "prop-types";
import {connect} from "react-redux";

export class AddProfile extends Component {
  state={
    firstname: '',
    lastname: '',
    email: '',
    role: '',
    image: '',
    password: '',
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.addNewProfile(this.state);
  };

  handleChange = (event) => {
    this.setState({[event.target.id]: event.target.value});
  };

  render() {
    return (
      <div>
        <h1>Add New User</h1>

        <div className="row clearfix products-grid">
          <div className="col-10 m-0-auto m-t-5">
            <form onSubmit={this.handleSubmit}>

              <div className="row">

                <div className="col-3">
                  <figure className="image-holder">
                    <img alt={'image'} src={require("../assets/images/profile/team-7.jpg")} />
                    <input type={'file'} id={'image'}
                           onChange={ (e) => this.handleChange({
                             target:{
                               id:'image',
                               value: e.target.files[0],
                             }
                           })
                           }/>

                  </figure>
                </div>


                <div className="col-7">
                  <table border="0" width="100%" className="product-single-table table-striped">
                    <tbody>
                    <tr>
                      <td><label htmlFor="firstname">First Name</label></td>
                      <td><input id="firstname" type="text" required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label htmlFor="lastname">Last Name</label></td>
                      <td><input id="lastname" type="text" required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label htmlFor="email">Email</label></td>
                      <td><input id="email" type="email" required={true} onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td><label>Role</label></td>
                      <td>
                        <select id={'role'} onChange={this.handleChange}>
                          <option>---Choose One---</option>
                          <option value='store_attendant'>Attendant</option>
                          <option value='admin'>Admin</option>
                        </select>
                      </td>
                    </tr>

                    <tr>
                      <td><label htmlFor="password">Password</label></td>
                      <td><input id="password" type="password" onChange={this.handleChange}/></td>
                    </tr>
                    <tr>
                      <td><label htmlFor="conf_password">Confirm Password</label></td>
                      <td><input id="conf_password" type="password" onChange={this.handleChange}/></td>
                    </tr>

                    <tr>
                      <td>&nbsp;</td>
                      <td><button type="submit" className="btn btn-primary">Save</button>
                        <button type="reset" className="btn btn-danger">Clear</button></td>
                    </tr>
                    </tbody>

                  </table>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    )
  }
}

AddProfile.propTypes = {
  addNewProfile: PropTypes.func.isRequired
};

export default connect(null, {
  addNewProfile,
})(AddProfile);
