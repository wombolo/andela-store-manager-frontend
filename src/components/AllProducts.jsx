import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getAllProducts, addToCart } from '../actions/productActions';
import {Link} from "react-router-dom";

export class AllProducts extends Component {
  componentWillMount() {
    this.props.getAllProducts();
  }

  addToCartEvt = (product) =>{
    this.props.addToCart(product);
  };

  render() {
    const {products} = this.props.products;

    if (products.length < 1){
      return(<p>Now Loading. Please be patient</p>)
    }


    return (
      <div>
        <h1>All Products</h1>

        <div className="row big-row products-row">
          {products.map(product => {
            const imageFile = require(`../assets/images/${product.image}`);

              return (
                <div key={product.id}>
                  <figure className="image-holder">
                    <Link to={`/edit-product/${product.id}`}>
                      <img src={imageFile} />
                    </Link>
                  </figure>
                  <div className="row">
                    <div className="col-3"><span className="pull-left item-price"> $ {product.price}</span></div>
                    <div className="col-6">

                      <button className="pull-right min-btn btn-primary" onClick={() => this.addToCartEvt(product)}>Add to Cart</button>
                    </div>
                  </div>
                  <h3>
                    <Link to={`/edit-product/${product.id}`}>
                    {product.title}</Link>
                  </h3>
                </div>
              )
            })
          }
        </div>

        <div className="row m-t-5 text-center">
          <h3 className='text-gold'>You may Scroll Right or Left</h3>
        </div>
        </div>
    )
  }
}

AllProducts.propTypes = {
  products: PropTypes.shape({}).isRequired,
  getAllProducts: PropTypes.func.isRequired,
  addToCart: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
  products: state.products,
});

export default connect(mapStateToProps, {
  getAllProducts,addToCart
})(AllProducts);

