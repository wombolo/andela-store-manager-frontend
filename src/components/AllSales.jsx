import React, { Component } from 'react';
import {connect} from "react-redux";
import { getAllSales} from "../actions/salesActions";
import {Link} from "react-router-dom";

export class AllSales extends Component {
  componentDidMount() {
    this.props.getAllSales();
  }

  render() {
    const {sales} = this.props.sales;
    const {role} = this.props.profile;

    let iteration = 0, grandTotalItems = 0, grandTotalAmount = 0;

    return (
      <div>
        <h1>My Sales</h1>

        <div className="row clearfix">
          <div className="col-10 m-0-auto m-t-5">

            <table className="dataTable table-striped" border="1" cellPadding="5">
              <thead>
              <tr>
                <td>S/N</td>
                <td>Transaction ID</td>
                <td>Item Name</td>
                <td>Item Price</td>
                <td>Item Quantity</td>
                <td>Amount</td>
                <td>Item Page</td>
                {role === 'admin'?<td> Attendant Profile </td>:null}
              </tr>
              </thead>

              <tbody>
              {sales.length > 0 ?
                sales.map(singleSale =>{
                  iteration++;

                  const thisAmountTotal = parseFloat((singleSale.price * singleSale.quantity).toFixed(2));
                  grandTotalItems += parseInt(singleSale.quantity);
                  grandTotalAmount += thisAmountTotal;

                  return(
                    <tr key={singleSale.id}>
                      <td>{iteration}</td>
                      <td><code>{Math.random().toString(36).substring(7).toUpperCase()}</code></td>
                      <td>{singleSale.title}</td>
                      <td>$ {singleSale.price}</td>
                      <td>{singleSale.quantity}</td>
                      <td>$ {thisAmountTotal}</td>
                      <td><Link to={`edit-product/${singleSale.product_id}`} className='btn btn-primary'>View</Link></td>

                      {role === 'admin'?
                        <td><Link to={`profile/${singleSale.profile_id}`} className='btn btn-dark'>View</Link></td>
                        : null
                      }

                    </tr>
                  )
                }) : <tr><td colSpan="7" className='center'>No Sales</td></tr>}
              </tbody>

              <tfoot>
              <tr>
                <td colSpan="4">Grand Total</td>
                <td>{grandTotalItems}</td>
                <td>$ {grandTotalAmount}</td>
                <td> </td>
                {role === 'admin'? <td> </td> : null}
              </tr>

              </tfoot>
            </table>

          </div>
        </div>

      </div>
    )
  }
}


export const mapStateToProps = state => ({
  sales: state.sales,
  profile: state.auth.user.profile
});


export default connect(mapStateToProps, {
  getAllSales
})(AllSales);
