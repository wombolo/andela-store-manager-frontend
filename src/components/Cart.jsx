import React, { Component } from 'react';
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {getAllCartItems, handleCheckout, removeFromStoreCart} from "../actions/productActions";

export class Cart extends Component {
  state = {};

  componentWillMount() {
    const {cart} = this.props;
    this.setState({cart});
  }

  handleQuantityChange = (event) =>{
    const itemId = parseInt(event.currentTarget.dataset.itemid);
    const itemValue = event.target.value;

    const cart = this.state.cart.splice(0);
    const position =  cart.findIndex((elem) => (elem.id === itemId ));
    cart[position].cartQty = itemValue;
    this.setState({cart})
  };

  handleRemoveFromCart = (id) =>{
    const cart = this.state.cart.splice(0);
    const position =  cart.findIndex((elem) => (elem.id === id ));
    cart.splice(position, 1);

    this.setState({cart: cart});
    this.props.removeFromStoreCart(cart);

  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.handleCheckout(this.state.cart);
  };

  render() {
    const {cart} = this.state;
    let iteration = 0, grandTotalItems = 0, grandTotalAmount = 0;

    return (
      <div>
        <h1>Checkout</h1>

        <form onSubmit={this.handleSubmit}>
          <div className="row clearfix">
            <div className="col-10 m-0-auto m-t-5">
              <h3>Cart </h3>

              <table className="dataTable table-striped" border="1" cellPadding="5">
                <thead>
                <tr>
                  <td>S/N</td>
                  <td>Transaction ID</td>
                  <td>Item Image</td>
                  <td>Item Name</td>
                  <td>Item Price</td>
                  <td>Item Quantity</td>
                  <td>Amount</td>
                  <td>Action</td>
                </tr>
                </thead>

                <tbody>
                {cart.length > 0 ?
                  cart.map(item =>{
                  const thisAmountTotal = parseFloat((item.price * item.cartQty).toFixed(2));

                  iteration++;
                  grandTotalItems += parseInt(item.cartQty);
                  grandTotalAmount += thisAmountTotal;

                  return(
                    <tr key={iteration}>
                    <td>{iteration}</td>
                    <td><code>{Math.random().toString(36).substring(7).toUpperCase()}</code></td>
                    <td><img src={require(`../assets/images/${item.image}`)} /></td>
                    <td>{item.title}</td>
                    <td>$ {item.price}</td>
                    <td><input type={'number'}
                               data-itemid={item.id}
                               defaultValue={item.cartQty}
                               max={item.cartQty + item.quantity}
                               onChange={this.handleQuantityChange} /></td>
                    <td>$ {thisAmountTotal}</td>
                    <td><button type='reset' id={`remove-${item.id}`} className='text-danger' onClick={()=> (this.handleRemoveFromCart(item.id))}>Remove</button></td>
                  </tr>
                  )
                })
                  : <tr><td colSpan="8" className='center'>No Items in Cart</td></tr>}
                </tbody>

                <tfoot>
                <tr>
                  <td colSpan="5">Grand Total</td>
                  <td>{grandTotalItems}</td>
                  <td>$ {grandTotalAmount}</td>
                  <td> </td>
                </tr>
                </tfoot>
              </table>
            </div>
          </div>

          {cart.length > 0 ?
            <div className="row clearfix">
              <div className="col-8"></div>
              <div className="col-2 pull-right">
                <button className="btn btn-success">Checkout</button>
              </div>
            </div>
            : ''
          }
        </form>
      </div>
    )
  }
}

Cart.propTypes = {
  getAllCartItems: PropTypes.func.isRequired,
  handleCheckout: PropTypes.func.isRequired,
  removeFromStoreCart: PropTypes.func.isRequired,
  cart: PropTypes.array.isRequired,
  profile: PropTypes.shape({}).isRequired,
};

export const mapStateToProps = state => ({
  cart: state.products.cart,
  profile: state.auth.user.profile
});

export default connect(mapStateToProps,{
  getAllCartItems, handleCheckout, removeFromStoreCart
})
(Cart);
