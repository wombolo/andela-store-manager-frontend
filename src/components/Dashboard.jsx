import React, { Component } from 'react';

export default class Dashboard extends Component {
  render() {
    return (
      <div>
        <h1>My Dashboard</h1>
      </div>
    )
  }
}
