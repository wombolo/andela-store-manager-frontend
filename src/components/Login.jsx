import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { loginUser } from '../actions/authActions';

/**
 * @param {function} event
 *  @returns {JSX} jsx
 */
export class Login extends Component {
  state = {
    email: '',
    password: '',
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const userData = {
      email: this.state.email,
      password: this.state.password,
    };
    this.props.loginUser(userData, this.props.history);
  };

  handleChange = (event) => {
    this.setState({ [event.target.id]: event.target.value });
  };

  /**
   * @param {function} render
   *  @returns {JSX} jsx
   */
  render() {
    return (

      <div className="container">
        <div className="row">
          <div className="col-5 m-0-auto m-t-15">

            <div className="logo">Vy Superstore</div>
            <div className="form-box">
              <h2 className="text-center">Login </h2>

              <form onSubmit={this.handleSubmit}>
                <div className="form-group">
                  <div className="col-3">
                    <label htmlFor="email">Email</label>
                  </div>
                  <div className="col-7">
                    <input className="form-control" type="email" id="email" onChange={this.handleChange} value={this.state.email}
                    />
                  </div>
                </div>

                <div className="form-group">
                  <div className="col-3">
                    <label htmlFor="password">Password</label>
                  </div>
                  <div className="col-7">
                    <input className="form-control" type="password" id="password" onChange={this.handleChange} />
                  </div>
                </div>

                <div className="form-group">
                  <div className="col-3">&nbsp;</div>
                  <div className="col-7">
                    <button
                      type='submit'
                      className="btn btn-primary"
                      disabled={this.props.auth.isLoading}>
                      {this.props.auth.isLoading ? 'Loading...' : 'Login'}
                    </button>
                    <button type="reset" className="btn btn-dark">Reset</button>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
Login.propTypes = {
  auth: PropTypes.shape({}).isRequired,
  history: PropTypes.shape({}).isRequired,
  loginUser: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps,
  { loginUser })(withRouter(Login));
