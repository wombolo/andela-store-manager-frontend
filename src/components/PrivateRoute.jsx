import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const PrivateRoute = ({ component: Component, auth, ...rest }) => (
    <Route
        {...rest}
        render={props => (auth.isAuthenticated ? (<Component {...props} />)
          : (<Redirect to='/login' />))}
    />
);

PrivateRoute.propTypes = {
  auth: PropTypes.object.isRequired,
  component: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  auth: state.auth,
});
export default connect(mapStateToProps)(PrivateRoute);
