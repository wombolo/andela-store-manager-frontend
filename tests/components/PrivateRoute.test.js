import 'babel-polyfill'
import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';

import { PrivateRoute } from '../../src/components/PrivateRoute.jsx';
import {Login} from "../../src/components/Login";
import {MemoryRouter} from "react-router-dom";

let wrapper;

const props = {
  component: () => Promise.all(),
  auth: {
    user: {
      profile: {
        id: 1,
        firstname: 'Akano',
        lastname: 'Peter',
        email: 'elkenzom1@gmail.com',
        role: 'admin',
        image: 'team-7.jpg',
        password: null
      },
      iat: 1551952568,
      exp: 1552038968
    },
    isAuthenticated: true,
    error: '',
    isLoading: false
  },
};


describe('PrivateRoute Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<PrivateRoute {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('Methods in Class', () => {

    beforeEach(() => {
      // store = createStore(combineReducers({}));
      wrapper = mount(<PrivateRoute {...props}/>);
    });

    afterEach(() => {
      wrapper.unmount();
    });
  });
});


describe('Methods in Class', () => {

  it('triggers when user not logged in', () => {
    props.auth.isAuthenticated = false;
    wrapper = mount(<MemoryRouter><PrivateRoute {...props}/></MemoryRouter>);
  });
});

