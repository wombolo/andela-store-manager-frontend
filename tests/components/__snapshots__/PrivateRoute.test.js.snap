// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Component should match snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
